stdlib/public/Cxx/CxxConvertibleToCollection.swift

```diff
@@ -21,36 +21,36 @@ public protocol CxxConvertibleToCollection {
   mutating func __endUnsafe() -> RawIterator
 }
 
-@inlinable @inline(__always)
-internal func forEachElement<C: CxxConvertibleToCollection>(
-  of c: C,
-  body: (C.RawIterator.Pointee) -> Void
-) {
-  var mutableC = c
-  withExtendedLifetime(mutableC) {
-    var rawIterator = mutableC.__beginUnsafe()
-    let endIterator = mutableC.__endUnsafe()
-    while rawIterator != endIterator {
-      body(rawIterator.pointee)
-      rawIterator = rawIterator.successor()
+extension CxxConvertibleToCollection {
+  @inlinable
+  internal func forEach(_ body: (RawIterator.Pointee) -> Void) {
+    var mutableSelf = self
+    withExtendedLifetime(mutableSelf) {
```

Comment:
[Note from the pessimistic little elf on my shoulder who has lost all trust in compilers]: There is some likelihood that this is going to be used in performance critical context, so it _may_ be a good idea to reorganize this so that the closure passed to `withExtendedLifetime` is empty:

```
var rawIterator = ...
let endIterator = ...
while rawIterator != endIterator { ... }
withExtendedLifetime(mutableSelf) {}
```

The compiler _usually_ has no trouble optimizing simple closures, but this may result in better code in some unusual cases.

Reply:
Fixed, thanks!